perf-tests/clusterloader2/pkg/measurement/util/object_store.go

```diff
@@ -146,13 +209,36 @@ func (s *PodStore) Stop() {
 	}
 }
 
-func FilterPods(pods []*v1.Pod, selector *ObjectSelector) []*v1.Pod {
+func FilterPods(ps *PodStore, selector *ObjectSelector) []*v1.Pod {
 	filteredPods := make([]*v1.Pod, 0)
+
+	// if has name label, use podLister func, otherwise, check every label
 	var selectorMap map[string]string
 	if selector.LabelSelector != "" {
 		selectorMap = getLabelSelectorMapFromString(selector.LabelSelector)
 	}
+	if ps.podListerFunc != nil {
+		labelKeyName := ""
+		labelKeyValue := ""
+		if name, isOK := selectorMap[labelNameKey]; isOK {
+			labelKeyName = labelNameKey
+			labelKeyValue = name
+		} else if group, isOK := selectorMap[labelGroupKey]; isOK {
+			labelKeyName = labelGroupKey
+			labelKeyValue = group
+		}
+		if labelKeyName != "" && labelKeyValue != "" {
+			pods, err := ps.podListerFunc(labelKeyName, labelKeyValue)
+			klog.Infof("==== FilterPods key [%v], value [%v], err [%v], len(pods)=[%v]", labelKeyName, labelKeyValue, err, len(pods))
```

Comment:
ho many log entries this generates ?

Reply:
about 2000 in 500 node test. Will delete before check in.

Reply:
Commented.